src/track/steminfoimporter.cpp

```diff
@@ -1,31 +1,57 @@
 #include "track/steminfoimporter.h"
 
+#include <QDataStream>
 #include <QFile>
 #include <QJsonArray>
 #include <QJsonDocument>
 #include <QJsonObject>
+#include <QMimeDatabase>
 #include <QString>
+#include <QtEndian>
+#include <QtGlobal>
 
 #include "util/logger.h"
 
-#define ATOM_TYPE(value)                                                   \
-    (uint32_t)(((uint8_t)(value)[0] << 24) | ((uint8_t)(value)[1] << 16) | \
-            ((uint8_t)(value)[2] << 8) | (uint8_t)(value)[3])
-
 namespace mixxx {
 
 namespace {
 
 const mixxx::Logger kLogger("StemInfoImporter");
 constexpr int kSupportedStemVersion = 1;
 const QStringList kStemFileExtensions = {".stem.mp4", ".stem.m4a"};
```

Comment:
As far as I learned, there are three cases:

- Anything that ends with `.mp4` is allowed
- `.stem.mp4` is recommended
- `.stem.m4a` is not specified, but is used in practice (Stemgen writes it and Traktor Pro reads it)

Reply:
Yes, that's correct and is already how it work. STEM extension are only defines here as fallback here in case the MIME detection fails from file content. Should I put a comment to explicitly mention that these extension are only used as fallback?

Reply:
Yes, a commet would help here!